src/ring_theory/ideal_operations.lean

```diff
@@ -304,6 +304,14 @@ theorem sup_mul : (I ⊔ J) * K = I * K ⊔ J * K :=
 submodule.sup_smul I J K
 variables {I J K}
 
+lemma pow_le_pow {m n : ℕ} (h : m ≤ n) :
+  I^n ≤ I^m :=
```

Comment:
Should this be called something with `mono` in ?

Reply:
I guess that would make sense... but there are a lot of other lemmas with `pow_le_pow` in their name in mathlib. @digama0 Which convention would you prefer?
@cipher1024 Is this something that should be tagged with `@[mono]` or something like that?

Reply:
Are ideals an ordered semiring?

Reply:
Good point, I'll look into proving it tomorrow. (I guess `submodule`s form an ordered semiring, and in the case of `ideal`s everything is `≤ 1`).

Reply:
@ChrisHughes24 Hmm, they are not.
https://github.com/leanprover-community/lean/blob/ec1613aef1eee72e601f192b16740629c6d49690/library/init/algebra/ordered_group.lean#L21 is not satisfied. Take any `I` and `J`, then `1 = I + 1 ≤ J + 1 = 1`, but we can't conclude `I ≤ J`.